cmd/common.go

```diff
@@ -107,7 +109,17 @@ func syncMain(ctx context.Context, filenames []string, dry bool, parallelism,
 	}
 
 	if targetContent.Info != nil {
-		dumpConfig.SelectorTags = targetContent.Info.SelectorTags
+		if len(targetContent.Info.SelectorTags) > 0 {
+			if len(dumpConfig.SelectorTags) > 0 {
+				sort.Strings(dumpConfig.SelectorTags)
+				sort.Strings(targetContent.Info.SelectorTags)
+				if !reflect.DeepEqual(dumpConfig.SelectorTags, targetContent.Info.SelectorTags) {
+					return fmt.Errorf("different tags specified by state file and --select-tags differ")
```

Comment:
```suggestion
					return fmt.Errorf("tags specified in the state file and via --select-tags flag are different. decK expects tags to be specified in either via flag or via state file. In case both are specified, they must match.")
```

Take care of line-wrapping.

@Tieske This cool?